src/Razor/src/Microsoft.VisualStudio.Editor.Razor/DefaultDynamicDocumentContainer.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.Razor.ProjectSystem;
+using Microsoft.CodeAnalysis.Razor.Workspaces;
+
+namespace Microsoft.CodeAnalysis.Razor
+{
+    // This types purpose is to serve as a non-Razor specific document delivery mechanism for Roslyn.
+    internal sealed class DefaultDynamicDocumentContainer : DynamicDocumentContainer
+    {
+        private readonly DocumentSnapshot _documentSnapshot;
+        private RazorDocumentExcerptService _excerptService;
+        private RazorSpanMappingService _mappingService;
+
+        public DefaultDynamicDocumentContainer(DocumentSnapshot documentSnapshot)
+        {
+            if (documentSnapshot is null)
+            {
+                throw new ArgumentNullException(nameof(documentSnapshot));
+            }
+
+            _documentSnapshot = documentSnapshot;
+        }
+
+        public override string FilePath => _documentSnapshot.FilePath;
+
+        public override TextLoader GetTextLoader(string filePath) => new GeneratedDocumentTextLoader(_documentSnapshot, filePath);
```

Comment:
So this returns the pre-set document snapshot regardless of the filePath that was provided, reporting it to be the requested path? I suspect I'm misunderstanding the purpose of this class, maybe a longer comment is in order? I would also make them doc-comments (even though they're not public) so that callers get the same context that's being provided in the source.

Reply:
So a `TextLoader` typically allows the retrieval of content under a specific file path. More specifically it's an asynchronous way to acquire a `SourceText` that has an associated file path with it.

This piece allows the generated content of our `DocumentSnapshot` (all our C# content) to be addressable by the provided filepath.